### PR TITLE
Added bypass for insert / delete for machines

### PR DESCRIPTION
**Pull request by felipeelia**
This is a "take over" of #1968, just to address some small things.

### Description of the Change

Allows indexing / deletion while in cron / WP CLI.

Props @nickdaugherty 
cc: @rinatkhaziev @pschoffer @netsuso @parkcityj

### Alternate Designs

N/A

### Benefits

Allows indexing / deletion while in cron / WP CLI.

### Possible Drawbacks

Allows indexing / deletion while in cron / WP CLI.

### Verification Process

We needed a way to do indexing / deletion while in cron / WP CLI. After this change, it worked.

### Checklist:

- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests passed.

### Applicable Issues

Closes #1968 .

### Changelog Entry

Added permission check bypass for indexing / deleting for cron and WP CLI. Props @nickdaugherty

### CODE REVIEW

**Comment by felipeelia** [comment]
@oscarssanchez do you mind giving a look at this? You've already worked in this part of the code, so probably you'll have some additional context to it. Thanks in advance!

File: includes/classes/Indexable/Post/SyncManager.php [hunk 1]
```diff
@@ -50,6 +50,31 @@ public function setup() {
 		add_action( 'added_post_meta', array( $this, 'action_queue_meta_sync' ), 10, 4 );
 		add_action( 'deleted_post_meta', array( $this, 'action_queue_meta_sync' ), 10, 4 );
 		add_action( 'wp_initialize_site', array( $this, 'action_create_blog_index' ) );
+
+		add_filter( 'ep_sync_insert_permissions_bypass', array( $this, 'filter_bypass_permission_checks_for_machines' ), 10, 2 );
```
Comment (oscarssanchez) [inline includes/classes/Indexable/Post/SyncManager.php:54]: As we are using just 1 arg, should we change it to 1 instead of 2 accepted args?

Comment (felipeelia) [reply includes/classes/Indexable/Post/SyncManager.php:54]: Good catch @oscarssanchez , thanks. I'll assign the PR back to me to change that :)

**Comment by felipeelia** [comment]
@oscarssanchez can you please give it another look? Thanks!

**Review by oscarssanchez — APPROVED** [review]
Looks good!

**Pull request merged**